### DISCUSSION TITLE
Question on Samba Hook - Move

### DISCUSSION THREAD

**Discussion by rajdallas** [apache/airflow#27875] [General]
Do we have a direct method implemented to move files at SMB file share level ? 

Currently I am accomplishing the move operation using shutil.copyfileobj and using Sambahook to remove the same ?

**Comment by Taragolis** [comment]
Seems like SambaHook just a thin wrapper around `smbprotocol` package

`SambaHook.replace` should work as move for file within the same SMB share, you could have a look docstring and implementation in [smbclient.replace](https://github.com/jborean93/smbprotocol/blob/1ea33abd58f2e3886220a10888608052bf3608c1/src/smbclient/_os.py#L503-L513)

https://github.com/apache/airflow/blob/b18f3190c6c157301cb7d29c439dddbb9a8dbf14/airflow/providers/samba/hooks/samba.py#L155-L157

**Reply by rajdallas** [reply]
Thank you. I did tried that option, however I see below error when I used replace or rename. All other method works fine for me.

```
  from airflow.providers.samba.hooks.samba import SambaHook
  share_hook = SambaHook(<Secret Manager>)
  fs_dir = 'source\\'
  arc_dir = 'archive\\'
  file = test.txt
  fs_hook.replace(fs_dir_name + file, arc_dir_name + file)
 
```

Exception:
 {{logging_mixin.py:109}} INFO - Exception : [Error 0] [NtStatus 0xc000019c] Unknown NtStatus error returned 'STATUS_FS_DRIVER_REQUIRED':  '//<ip address>/share_name/archive\test.txt'

**Reply by Taragolis** [reply]
Im not familiar with this hook but seems like it intend to use as context manager 
```python
from airflow.providers.samba.hooks.samba import SambaHook

fs_dir = 'source\\'
arc_dir = 'archive\\'
file = test.txt

with SambaHook(<Secret Manager>) as fs_hook:
    fs_hook.replace(fs_dir_name + file, arc_dir_name + file)
```

**Reply by rajdallas** [reply]
It didn't help me either.

**Reply by Taragolis** [reply]
You could also try to build your path with `\` (forward slash) rather than `/` backslash. Example:

```python
>>> file = "file.txt"
>>> fr"\\share_name\source\{file}"
'\\\\share_name\\source\\file.txt'
```

It is just my assumption. If it not help you should try to debug by yourself I would recommend try to use [smbprotocol](https://github.com/jborean93/smbprotocol) package directly and try to check is this error gone or not. 
If not than issue not in Airflow provider.

And also try to check this resources related to SMB
- https://github.com/jborean93/smbprotocol/issues?q=is%3Aissue+STATUS_FS_DRIVER_REQUIRED
- https://davidvielmetter.com/tips/ntstatus-error-code-list/
